lib/rules/no-forbidden-elements.js

```diff
@@ -0,0 +1,90 @@
+'use strict';
+
+const Rule = require('./base');
+const createErrorMessage = require('../helpers/create-error-message');
+
+function ERROR_MESSAGE_FORBIDDEN_ELEMENTS(element) {
+  return `Use of <${element}> detected. Do not use forbidden elements.`;
+}
+
+const FORBIDDEN_ELEMENTS = ['meta', 'style', 'html', 'script'];
+
+const DEFAULT_CONFIG = {
+  forbidden: [],
+};
+
+function isValidConfigObjectFormat(config) {
+  for (let key in config) {
+    let value = config[key];
+    let valueIsArray = Array.isArray(value);
+
+    if (key === 'forbidden' && !valueIsArray) {
+      return false;
+    } else if (!DEFAULT_CONFIG[key]) {
+      return false;
+    }
+  }
+
+  return true;
+}
+
+function sanitizeConfigArray(forbidden = []) {
+  return forbidden.filter((option) => option !== '').sort((a, b) => b.length - a.length);
+}
+
+module.exports = class NoForbiddenElements extends Rule {
+  parseConfig(config) {
+    let configType = typeof config;
+
+    switch (configType) {
+      case 'boolean':
+        // if `true` use `DEFAULT_CONFIG`
+        return config ? DEFAULT_CONFIG : false;
+      case 'object':
+        if (Array.isArray(config)) {
+          return {
+            forbidden: sanitizeConfigArray(config),
+          };
+        } else if (isValidConfigObjectFormat(config)) {
+          // default any missing keys to empty values
+          return {
+            forbidden: sanitizeConfigArray(config.forbidden || []),
+          };
+        }
+        break;
+      case 'undefined':
+        return false;
+    }
+
+    let errorMessage = createErrorMessage(
+      this.ruleName,
+      [
+        '  * boolean - `true` to enable / `false` to disable',
+        '  * array -- an array of strings to forbid',
+        '  * object -- An object with the following keys:',
+        '    * `forbidden` -- An array of forbidden strings',
+      ],
+      config
+    );
+
+    throw new Error(errorMessage);
+  }
+  visitor() {
+    return {
+      ElementNode(node) {
+        let isForbiddenElement =
+          FORBIDDEN_ELEMENTS.includes(node.tag) || this.config.forbidden.includes(node.tag);
+        if (isForbiddenElement) {
```

Comment:
We should not hard code `FORBIDDEN_ELEMENTS` like this, `this.config.forbidden` will include the `FORBIDDEN_ELEMENTS` by default (when using `true` for config value) but if configured the explicit list that is provided should be used. 

tldr;

```suggestion
        let isForbiddenElement = this.config.forbidden.includes(node.tag);
        if (isForbiddenElement) {
```

Reply:
Done